### PR TITLE
Feature/e2etest

### PR DESCRIPTION
**Pull request by mehab**
End to end test for completing issue: https://github.com/DependencyTrack/hyades/issues/573

### CODE REVIEW

File: e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java [hunk 1]
```diff
@@ -225,4 +239,51 @@ private void verifyWebhookNotification() {
...
+    private void verifyProjectVulnAnalysisCompleteNotification() {
+        wireMock.verify(postRequestedFor(urlPathEqualTo("/notification"))
+                .withRequestBody(equalToJson("""
+                        {
+                           "notification" : {
+                             "level" : "LEVEL_INFORMATIONAL",
+                             "scope" : "SCOPE_PORTFOLIO",
+                             "group" : "GROUP_PROJECT_VULN_ANALYSIS_COMPLETE",
+                             "timestamp" : "${json-unit.any-string}",
+                             "title" : "Project vulnrability analysis complete",
```
Comment (nscuro) [inline e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java:251]: Typo in `vulnrability`.

File: e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java [hunk 2]
```diff
@@ -225,4 +239,51 @@ private void verifyWebhookNotification() {
...
+                             "content" : "${json-unit.any-string}",
+                             "subject" : {
+                               "project" : {
+                                 "uuid": "${json-unit.any-string}",
+                                 "name" : "foo",
+                                 "version" : "bar"
+                               },
+                               "componentAnalysisComplete" : [ {
```
Comment (nscuro) [inline e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java:259]: The naming is a little odd as it doesn't really describe what the array is representing. Perhaps something like `findings` would be more fitting?

File: e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java [hunk 3]
```diff
@@ -225,4 +239,51 @@ private void verifyWebhookNotification() {
...
+                                   "sha256" : "efb86b148712a838b94b3cfc95769785a116b3461f709b4cc510055a58b804b2",
+                                   "sha512" : "0e9398591d86f80f16fc2d6ff0dda3e7821033e2c59472981eaab61443be3d77198655682905b85260fb2186a2cf0f33988aff689a49bb54e56c07e02f607e8a"
+                                 },
+                                 "vulnerability" : [ {
+                                   "uuid": "${json-unit.any-string}",
+                                   "vulnId" : "INT-123",
+                                   "source" : "INTERNAL",
+                                   "title" : "NA",
+                                   "subTitle" : "NA",
+                                   "recommendation" : "NA",
```
Comment (nscuro) [inline e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java:277]: Was there a requirement for these fields to always be present? If there is no value, I would've expected the fields to not be present at all.

**Comment by mehab** [comment]
All the changes added in this PR: https://github.com/DependencyTrack/hyades-apiserver/pull/204/files and https://github.com/DependencyTrack/hyades/pull/620 Reason for creating a separate PR is end to end test cannot be tested without the proto changes merged.

File: e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java (hunk 3 above)
Comment (mehab) [reply e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java]: fixed :)

File: e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java (hunk 2 above)
Comment (mehab) [reply e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java]: fixed :)

File: e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java (hunk 1 above)
Comment (mehab) [reply e2e/src/test/java/org/hyades/e2e/BomUploadProcessingE2ET.java]: fixed

**Review by nscuro — APPROVED** [review]

**Pull request merged**